wordpress-mobile/WordPress-iOS
# Stats: shared UI elements for stats cards

Ref #10466 

**Details:**

This basically lays the foundation needed to display `All Time Stats` by adding common views used to build up a stats card:

- `StatsCellHeader` that shows a label and the ellipsis.

<img width="350" alt="cell_header" src="https://user-images.githubusercontent.com/1816888/48655162-f2285900-e9d0-11e8-9995-d5e3f3a56198.png">

- `StatsTotalRow` to show a data row on a card. Includes all possible UI elements, where optional items can be toggled.

<img width="350" alt="row" src="https://user-images.githubusercontent.com/1816888/48655197-492e2e00-e9d1-11e8-946e-55941affc8c0.png">
<img width="350" alt="row2" src="https://user-images.githubusercontent.com/1816888/48655200-4df2e200-e9d1-11e8-970f-b6afbede8630.png">

- `SimpleTotalsCell` to show a stat card that only displays totals (i.e. has no actions, just data rows).

<img width="350" alt="all_time_stats" src="https://user-images.githubusercontent.com/1816888/48655219-77ac0900-e9d1-11e8-93b7-c3ea8054a98c.png">

**Notes:**
- There are a few `var showBlah` properties used to toggle UI elements, especially in `StatsTotalRow`. These will most likely go away when I create `configure` methods. For now, these just help with testing. So don't get too attached to/offended by them.
- I have added a stub `All Time Stats` card to Insights. But right now it only serves to example the `SimpleTotalsCell` and various `StatsTotalRow` states. It will become `All Time Stats` in the next PR.

**What's missing:**
- `No data yet` view.
- Action for the row disclosure.

---

**To test:**
There are two _temporary_ methods in `SimpleTotalsCell` to example what's been done.

1. `addSimpleTotalsRows`: this examples what a simple total stat card will look like (aka `All Time Stats`, our end goal). This is enabled by default. It will show:

<img width="350" alt="simple_totals" src="https://user-images.githubusercontent.com/1816888/48655297-57c91500-e9d2-11e8-8682-3ec53b9ac90f.png">

2. `addExampleRows`: this examples the various states of a cell row by toggling the different UI elements. If you care to see this in action, enable this method and disable `addSimpleTotalsRows`. It will show:

<img width="350" alt="misc_rows" src="https://user-images.githubusercontent.com/1816888/48655339-c017f680-e9d2-11e8-9bd3-59fe1a4bb916.png">
